aip/general/0236.md

```diff
@@ -0,0 +1,377 @@
+- Cascading deletes **must** occur: if the top-level resource is deleted, all
+  resources underlying any experiments must also be deleted.
+- The etag on the experiment **must** guidance on the resource freshness
+  validation ([AIP-154][aip-154]) and will be a randomly generated UUID
+  ([AIP-148][aip-148]). The etag **must** be present in the preview logs.
+  - The parent resource **must** have an `etag` as well.
+  - Both etags **must** be present in the preview logs to uniquely identify the
+    version of the live and experimental policies. These will change when either
+    Policy configurations change.
+
+
+### Metadata
+A `metadata` field is used to track all variables pertinent to a preview.
+ 
+```proto 
+message PreviewMetadata {
+  // State of the previewed resource. Possible values are ACTIVE and INACTIVE, the 
+  // former which indicates that results are being logged downstream.
+  string state = 1;
+
+  // Ifentifying string common to all preview mode logs for the product. Searching 
+  // logs for this string will isolate all logs pertaining to this preview.
+  string log_prefix = 2;
+
+  // The start time of the current preview.
+  google.protobuf.Timestamp start_time = 3;
+
+  // The latest stop time of the preview.
+  google.protobuf.Timestamp stop_time = 4;
+}
+```
+- `PreviewMetadata` **must** have the fields defined in the proto above and
+  **must** use the same set of values for state.
+- A shared proto used across services **must not** be used ([AIP-213][aip-213]).
+- When a preview is first started, preview_metadata **must** be empty. It will
+  only be present on experiments once a custom method has been used to start it.
+- All preview metadata fields **must** be output only.
+- The state field **must** only be updated by the service:
+  - The state **must** subsequently be altered between ACTIVE and INACTIVE to
+    start or stop downstream evaluation using custom methods described below.
+- The first time the startPreview custom method is used, the service **must**
+  create the `PreviewMetadata` and do the following: 
+  - It **must** set the `state` to ACTIVE
+  - It **must** set a `log_prefix` which is common to all preview mode logs for
+    the product.
+  - It **must** populate the `start_time` with the current time. 
+- When the stopPreview custom method is used, the service **must** do the
+  following:
+  - It **must** set the `state` to INACTIVE
+  - It **must** populate the `stop_time` with the current time. 
+- `start_time` **must** be updated every time the status is changed to ACTIVE.
+
+### Collection extensibility
+
+For the collection use-case, Rule is a repeated field on a Policy resource as
+shown below.
+
+```proto
+message Policy {
+
+  string name = 1;
+
+  repeated Rule rules = 2;
+
+  // Other existing fields…
+}
+```
+
+An experiment is created for the entire parent resource. So essentially, there
+will be a experiment with its own set of rules. The following guidelines
```

Comment:
You say, "The following guidelines...". Which are you referring to? After this sentence there is no colon. You go directly into examples and not "guidelines". I find this confusing.